contracts/chainlink/OrderAPIConsumer.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.6;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/utils/Address.sol';
+import '@chainlink/contracts/src/v0.6/ChainlinkClient.sol';
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+interface IOrderAPIConsumer {
+    /**
+     * Call the rendering API with the address of the requestor
+     */
+    function executeRequest(address requestor) external returns (bytes32 requestId);
+}
+
+interface IOrderAPICallback {
+    /**
+     * Call the callback function with the job id and the result (usually the ipfs hash)
+     */
+    function fulfillResponse(bytes32 requestId, bytes32 result) external;
+}
+
+interface IOrderAPIConsumerAdmin {
+    /**
+     * set the callback address used by the consumer
+     */
+    function setCallback(address callback) external;
+
+    /**
+     * Set the job id
+     */
+    function setJobId(string memory jobId) external;
+
+    /**
+     * Set the fee for executing the job
+     */
+    function setFee(uint256 fee) external;
+}
+
+contract OrderAPIConsumer is Ownable, ChainlinkClient, IOrderAPIConsumer, IOrderAPIConsumerAdmin {
+    using Address for address;
+
+    IOrderAPICallback private _callback;
+    bytes32 private _jobId;
+    uint256 private _fee;
+
+    event CallbackUpdated(address previous, address current);
+    event JobIdUpdated(bytes32 previous, bytes32 current);
+    event FeeUpdated(uint256 previous, uint256 current);
+
+    event ResponseFulfilled(bytes32 requestId, bytes32 result);
+
+    constructor(
+        address link,
+        address oracle,
+        address callback,
+        string memory jobId,
+        uint256 fee
+    ) public Ownable() {
+        setChainlinkToken(link);
+        setChainlinkOracle(oracle);
+
+        if (callback != address(0)) {
+            _callback = IOrderAPICallback(callback);
+        }
+        setJobId(jobId);
+        _fee = fee; //0.1 * 10**18; // 0.1 LINK
+    }
+
+    // IOrderAPIConsumer
+
+    function executeRequest(address requestor) public override returns (bytes32 requestId) {
```

Comment:
i left this function open for now for testing.  To test the API integration you do not have to deploy the rarepizzas contract.  if you just set the callback the the zero address, you can call this function directly and get an event posted back with the truncated ipfs hash.  useful for debugging.